--- /dev/null
+++ b/third_party/ublock/BUILD.gn
@@ -0,0 +1,50 @@
+# Copyright 2025 The Helium Authors
+# You can use, redistribute, and/or modify this source code under
+# the terms of the GPL-3.0 license that can be found in the LICENSE file.
+
+import("//ui/webui/resources/tools/generate_grd.gni")
+import("//tools/grit/grit_rule.gni")
+
+file_list = "$target_gen_dir/files.json"
+generated_grd = "$target_gen_dir/ublock_resources.grd"
+
+action("generate_file_list") {
+    script = "generate_file_list.py"
+    args = [
+        rebase_path("."),
+        rebase_path(file_list),
+    ]
+    inputs = [
+        "manifest.json",
+        "generate_file_list.py",
+    ]
+    outputs = [ file_list ]
+}
+
+generate_grd("ublock_grd") {
+    out_grd = generated_grd
+
+    grd_prefix = "ublock"
+    resource_path_prefix = "ublock"
+
+    manifest_files = [ file_list ]
+    deps = [":generate_file_list"]
+}
+
+grit("ublock") {
+    source = generated_grd
+
+    # Required because the .grd is generated.
+    enable_input_discovery_for_gn_analyze = false
+
+    output_dir = "$target_gen_dir/resources"
+
+    outputs = [
+        "grit/ublock_resources.h",
+        "grit/ublock_resources_map.cc",
+        "grit/ublock_resources_map.h",
+        "ublock_resources.pak"
+    ]
+
+    deps = [":ublock_grd"]
+}
--- /dev/null
+++ b/third_party/ublock/generate_file_list.py
@@ -0,0 +1,19 @@
+# Copyright 2025 The Helium Authors
+# You can use, redistribute, and/or modify this source code under
+# the terms of the GPL-3.0 license that can be found in the LICENSE file.
+
+import json
+from sys import argv
+from pathlib import Path
+
+ublock_src = argv[1]
+out_file = argv[2]
+
+with open(out_file, 'w') as f:
+  f.write(json.dumps(
+    { "base_dir": ublock_src,
+      "files":   [ str((root / file).relative_to(ublock_src))
+                   for root, _, files in Path(ublock_src).walk()
+                   for file in files ]
+    }
+  ))
--- a/tools/gritsettings/resource_ids.spec
+++ b/tools/gritsettings/resource_ids.spec
@@ -196,7 +196,7 @@
     "META": {"sizes": {"includes": [500]}},
     "includes": [2850],
   },
-  "<(SHARED_INTERMEDIATE_DIR)/chrome/browser/resources/ash/settings/resources.grd": {
+  "<(SHARED_INTERMEDIATE_DIR)/third_party/ublock/ublock_resources.grd": {
     "META": {"sizes": {"includes": [1000],}},
     "includes": [2860],
   },
--- a/chrome/browser/extensions/BUILD.gn
+++ b/chrome/browser/extensions/BUILD.gn
@@ -595,6 +595,7 @@ source_set("extensions") {
     "//net:extras",
     "//services/metrics/public/cpp:ukm_builders",
     "//storage/browser",
+    "//third_party/ublock",
     "//third_party/zlib/google:zip",
     "//ui/base/accelerators/global_accelerator_listener",
     "//ui/base/unowned_user_data",
--- a/chrome/chrome_paks.gni
+++ b/chrome/chrome_paks.gni
@@ -151,6 +151,7 @@ template("chrome_extra_paks") {
       "$root_gen_dir/third_party/blink/public/resources/blink_resources.pak",
       "$root_gen_dir/third_party/blink/public/resources/inspector_overlay_resources.pak",
       "$root_gen_dir/third_party/blink/public/strings/permission_element_generated_strings.pak",
+      "$root_gen_dir/third_party/ublock/resources/ublock_resources.pak",
       "$root_gen_dir/ui/resources/ui_lottie_resources.pak",
       "$root_gen_dir/ui/webui/resources/webui_resources.pak",
     ]
@@ -173,6 +174,7 @@ template("chrome_extra_paks") {
       "//third_party/blink/public:devtools_inspector_resources",
       "//third_party/blink/public:resources",
       "//third_party/blink/public/strings:permission_element_generated_strings",
+      "//third_party/ublock",
       "//ui/resources:ui_lottie_resources_grd",
       "//ui/webui/resources",
     ]
--- a/chrome/browser/extensions/component_loader.h
+++ b/chrome/browser/extensions/component_loader.h
@@ -215,6 +215,7 @@ class ComponentLoader : public KeyedServ
                                  const std::string& name_string,
                                  const std::string& description_string);
   void AddWebStoreApp();
+  void AddUBlock();
 
 #if BUILDFLAG(IS_CHROMEOS)
   void AddChromeApp();
--- a/chrome/browser/extensions/component_loader.cc
+++ b/chrome/browser/extensions/component_loader.cc
@@ -57,6 +57,7 @@
 #include "ui/accessibility/accessibility_features.h"
 #include "ui/base/l10n/l10n_util.h"
 #include "ui/base/resource/resource_bundle.h"
+#include "third_party/ublock/resources/grit/ublock_resources.h"
 
 #if BUILDFLAG(IS_CHROMEOS)
 #include "ash/constants/ash_features.h"
@@ -429,6 +430,21 @@ void ComponentLoader::AddWebStoreApp() {
       l10n_util::GetStringUTF8(IDS_WEBSTORE_APP_DESCRIPTION));
 }
 
+void ComponentLoader::AddUBlock() {
+  Add(IDR_UBLOCK_MANIFEST_JSON, base::FilePath(FILE_PATH_LITERAL("ublock")));
+
+  using namespace extension_l10n_util;
+  ComponentExtensionInfo& info = component_extensions_.back();
+  std::string error;
+  bool localized = LocalizeExtension(info.root_directory, &info.manifest,
+                      GzippedMessagesPermission::kAllowForTrustedSource,
+                      &error, true);
+
+  if (!localized) {
+    LOG(ERROR) << "Failed to localize uBlock: " << error;
+  }
+}
+
 #if BUILDFLAG(IS_CHROMEOS)
 void ComponentLoader::AddChromeApp() {
   AddWithNameAndDescription(
@@ -506,6 +522,7 @@ void ComponentLoader::AddDefaultComponen
 
   if (!skip_session_components) {
     AddWebStoreApp();
+    AddUBlock();
 #if BUILDFLAG(IS_CHROMEOS)
     AddChromeApp();
 #endif  // BUILDFLAG(IS_CHROMEOS)
--- a/chrome/browser/extensions/component_extensions_allowlist/allowlist.cc
+++ b/chrome/browser/extensions/component_extensions_allowlist/allowlist.cc
@@ -15,8 +15,10 @@
 #include "chrome/common/buildflags.h"
 #include "chrome/common/extensions/extension_constants.h"
 #include "chrome/grit/browser_resources.h"
+#include "components/helium_services/extension_ids.h"
 #include "extensions/common/constants.h"
 #include "printing/buildflags/buildflags.h"
+#include "third_party/ublock/resources/grit/ublock_resources.h"
 
 #if BUILDFLAG(IS_CHROMEOS)
 #include "ash/keyboard/ui/grit/keyboard_resources.h"
@@ -31,6 +33,7 @@ bool IsComponentExtensionAllowlisted(con
   constexpr auto kAllowed = base::MakeFixedFlatSet<std::string_view>({
       extension_misc::kInAppPaymentsSupportAppId,
       extension_misc::kPdfExtensionId,
+      helium::kUBlockOriginComponentId,
 #if BUILDFLAG(IS_CHROMEOS)
       extension_misc::kAssessmentAssistantExtensionId,
       extension_misc::kAccessibilityCommonExtensionId,
@@ -86,6 +89,7 @@ bool IsComponentExtensionAllowlisted(int
     case IDR_NETWORK_SPEECH_SYNTHESIS_MANIFEST_MV3:
     case IDR_READING_MODE_GDOCS_HELPER_MANIFEST:
     case IDR_WEBSTORE_MANIFEST:
+    case IDR_UBLOCK_MANIFEST_JSON:
 
 #if BUILDFLAG(IS_CHROMEOS)
     // Separate ChromeOS list, as it is quite large.
--- a/chrome/browser/extensions/chrome_component_extension_resource_manager.cc
+++ b/chrome/browser/extensions/chrome_component_extension_resource_manager.cc
@@ -24,6 +24,7 @@
 #include "extensions/common/extension_id.h"
 #include "pdf/buildflags.h"
 #include "ui/base/resource/resource_bundle.h"
+#include "third_party/ublock/resources/grit/ublock_resources_map.h"
 
 #if BUILDFLAG(IS_CHROMEOS)
 #include "ash/keyboard/ui/resources/keyboard_resource_util.h"
@@ -102,6 +103,7 @@ ChromeComponentExtensionResourceManager:
 
   AddComponentResourceEntries(kComponentExtensionResources);
   AddComponentResourceEntries(kExtraComponentExtensionResources);
+  AddComponentResourceEntries(kUblockResources);
 
 #if BUILDFLAG(IS_CHROMEOS)
   // Add Files app JS modules resources.
--- a/ui/base/resource/resource_bundle.cc
+++ b/ui/base/resource/resource_bundle.cc
@@ -768,7 +768,7 @@ base::RefCountedMemory* ResourceBundle::
 
   std::string_view data = GetRawDataResourceForScale(resource_id, scale_factor);
   if (data.empty())
-    return nullptr;
+    return new base::RefCountedString();
 
   if (net::GZipHeader::HasGZipHeader(base::as_byte_span(data)) ||
       HasBrotliHeader(data)) {
--- a/extensions/common/extension_l10n_util.h
+++ b/extensions/common/extension_l10n_util.h
@@ -81,7 +81,8 @@ bool LocalizeManifest(const extensions::
 bool LocalizeExtension(const base::FilePath& extension_path,
                        base::Value::Dict* manifest,
                        GzippedMessagesPermission gzip_permission,
-                       std::string* error);
+                       std::string* error,
+                       bool is_component = false);
 
 // Adds locale_name to the extension if it's in chrome_locales, and
 // if messages file is present (we don't check content of messages file here).
@@ -129,6 +130,11 @@ extensions::MessageBundle* LoadMessageCa
     GzippedMessagesPermission gzip_permission,
     std::string* error);
 
+extensions::MessageBundle* LoadComponentMessageCatalogs(
+    const base::FilePath& extension_root,
+    const std::string& default_locale,
+    std::string* error);
+
 // Loads message catalogs for all locales to check for validity. Used for
 // validating unpacked extensions.
 bool ValidateExtensionLocales(const base::FilePath& extension_path,
--- a/extensions/common/extension_l10n_util.cc
+++ b/extensions/common/extension_l10n_util.cc
@@ -18,14 +18,18 @@
 #include "base/files/file_enumerator.h"
 #include "base/files/file_util.h"
 #include "base/json/json_file_value_serializer.h"
+#include "base/json/json_string_value_serializer.h"
 #include "base/json/json_reader.h"
 #include "base/logging.h"
+#include "base/memory/ref_counted_memory.h"
 #include "base/no_destructor.h"
 #include "base/strings/strcat.h"
 #include "base/strings/string_util.h"
 #include "base/strings/stringprintf.h"
 #include "base/strings/utf_string_conversions.h"
 #include "base/values.h"
+#include "extensions/browser/component_extension_resource_manager.h"
+#include "extensions/browser/extensions_browser_client.h"
 #include "extensions/common/constants.h"
 #include "extensions/common/error_utils.h"
 #include "extensions/common/extension.h"
@@ -38,6 +42,7 @@
 #include "third_party/icu/source/common/unicode/uloc.h"
 #include "third_party/zlib/google/compression_utils.h"
 #include "ui/base/l10n/l10n_util.h"
+#include "ui/base/resource/resource_bundle.h"
 
 namespace errors = extensions::manifest_errors;
 namespace keys = extensions::manifest_keys;
@@ -113,6 +118,19 @@ std::optional<base::Value::Dict> LoadMes
   return dictionary;
 }
 
+std::optional<base::Value::Dict> LoadMessageFile(
+    int resource_id, std::string* error) {
+  const ui::ResourceBundle& rb = ui::ResourceBundle::GetSharedInstance();
+  JSONStringValueDeserializer messages_deserializer(
+      base::as_string_view(*rb.LoadDataResourceBytes(resource_id)));
+
+  if (auto value = messages_deserializer.Deserialize(nullptr, error); value) {
+    return std::move(*value).TakeDict();
+  }
+
+  return std::nullopt;
+}
+
 // Localizes manifest value of string type for a given key.
 bool LocalizeManifestValue(const std::string& key,
                            const extensions::MessageBundle& messages,
@@ -360,13 +378,16 @@ bool LocalizeManifest(const extensions::
 bool LocalizeExtension(const base::FilePath& extension_path,
                        base::Value::Dict* manifest,
                        GzippedMessagesPermission gzip_permission,
-                       std::string* error) {
+                       std::string* error,
+                       bool is_component) {
   DCHECK(manifest);
 
   std::string default_locale = GetDefaultLocaleFromManifest(*manifest, error);
 
   std::unique_ptr<extensions::MessageBundle> message_bundle(
-      extensions::file_util::LoadMessageBundle(extension_path, default_locale,
+      is_component ? extensions::file_util::LoadComponentMessageBundle(extension_path,
+                                                                default_locale, error)
+      : extensions::file_util::LoadMessageBundle(extension_path, default_locale,
                                                gzip_permission, error));
 
   if (!message_bundle && !error->empty())
@@ -504,6 +525,42 @@ extensions::MessageBundle* LoadMessageCa
   return extensions::MessageBundle::Create(catalogs, error);
 }
 
+extensions::MessageBundle* LoadComponentMessageCatalogs(
+    const base::FilePath& extension_root,
+    const std::string& default_locale,
+    std::string* error) {
+  std::vector<std::string> all_fallback_locales;
+  base::FilePath locale_path(extensions::kLocaleFolder);
+
+  auto* manager =
+      extensions::ExtensionsBrowserClient::Get()->GetComponentExtensionResourceManager();
+
+  GetAllFallbackLocales(default_locale, &all_fallback_locales);
+
+  extensions::MessageBundle::CatalogVector catalogs;
+  for (const auto& fallback_locale : all_fallback_locales) {
+    base::FilePath this_locale_path =
+      locale_path.AppendASCII(fallback_locale)
+                 .AppendASCII(extensions::kMessagesFilename);
+    int resource_id = 0;
+
+    if (!manager->IsComponentExtensionResource(
+          extension_root, this_locale_path, &resource_id)) {
+      continue;
+    }
+
+    std::optional<base::Value::Dict> catalog = LoadMessageFile(resource_id, error);
+    if (!catalog.has_value()) {
+      return nullptr;
+    }
+
+    catalogs.push_back(std::move(*catalog));
+  }
+
+  return extensions::MessageBundle::Create(
+           catalogs, error);
+}
+
 bool ValidateExtensionLocales(const base::FilePath& extension_path,
                               const base::Value::Dict& manifest,
                               std::string* error) {
--- a/extensions/browser/l10n_file_util.h
+++ b/extensions/browser/l10n_file_util.h
@@ -47,7 +47,8 @@ LoadMessageBundleSubstitutionMapFromPath
     const std::vector<base::FilePath>& paths,
     const ExtensionId& extension_id,
     const std::string& default_locale,
-    extension_l10n_util::GzippedMessagesPermission gzip_permission);
+    extension_l10n_util::GzippedMessagesPermission gzip_permission,
+    bool is_component = false);
 
 }  // namespace extensions::l10n_file_util
 
--- a/extensions/browser/l10n_file_util.cc
+++ b/extensions/browser/l10n_file_util.cc
@@ -38,7 +38,8 @@ LoadMessageBundleSubstitutionMapFromPath
     const std::vector<base::FilePath>& paths,
     const ExtensionId& extension_id,
     const std::string& default_locale,
-    extension_l10n_util::GzippedMessagesPermission gzip_permission) {
+    extension_l10n_util::GzippedMessagesPermission gzip_permission,
+    bool is_component) {
   std::unique_ptr<MessageBundle::SubstitutionMap> return_value =
       LoadNonLocalizedMessageBundleSubstitutionMap(extension_id);
 
@@ -49,8 +50,13 @@ LoadMessageBundleSubstitutionMapFromPath
 
   std::string error;
   for (const base::FilePath& path : paths) {
-    std::unique_ptr<MessageBundle> bundle(file_util::LoadMessageBundle(
-        path, default_locale, gzip_permission, &error));
+    std::unique_ptr<MessageBundle> bundle(
+        is_component
+        ? file_util::LoadComponentMessageBundle(
+            path, default_locale, &error)
+        : file_util::LoadMessageBundle(
+            path, default_locale, gzip_permission, &error));
+
     if (bundle) {
       for (const auto& iter : *bundle->dictionary()) {
         // |insert| only adds new entries, and does not replace entries in
--- a/extensions/common/file_util.h
+++ b/extensions/common/file_util.h
@@ -162,6 +162,11 @@ MessageBundle* LoadMessageBundle(
     extension_l10n_util::GzippedMessagesPermission gzip_permission,
     std::string* error);
 
+MessageBundle* LoadComponentMessageBundle(
+      const base::FilePath& extension_root,
+      const std::string& default_locale,
+      std::string* error);
+
 // Helper functions for getting paths for files used in content verification.
 base::FilePath GetVerifiedContentsPath(const base::FilePath& extension_path);
 base::FilePath GetComputedHashesPath(const base::FilePath& extension_path);
--- a/extensions/common/file_util.cc
+++ b/extensions/common/file_util.cc
@@ -593,6 +593,25 @@ MessageBundle* LoadMessageBundle(
   return message_bundle;
 }
 
+MessageBundle* LoadComponentMessageBundle(
+      const base::FilePath& extension_root,
+      const std::string& default_locale,
+      std::string* error) {
+  error->clear();
+
+  std::set<std::string> chrome_locales;
+  extension_l10n_util::GetAllLocales(&chrome_locales);
+
+  if (default_locale.empty() || !base::Contains(chrome_locales, default_locale)) {
+    *error = l10n_util::GetStringUTF8(
+        IDS_EXTENSION_LOCALES_NO_DEFAULT_LOCALE_SPECIFIED);
+    return nullptr;
+  }
+
+  return extension_l10n_util::LoadComponentMessageCatalogs(
+      extension_root, default_locale, error);
+}
+
 base::FilePath GetVerifiedContentsPath(const base::FilePath& extension_path) {
   return extension_path.Append(kMetadataFolder)
       .Append(kVerifiedContentsFilename);
--- a/extensions/browser/renderer_startup_helper.cc
+++ b/extensions/browser/renderer_startup_helper.cc
@@ -608,6 +608,18 @@ void RendererStartupHelper::GetMessageBu
     paths_to_load.push_back(imported_extension->path());
   }
 
+  bool is_component =
+    extension->location() == extensions::mojom::ManifestLocation::kComponent;
+
+  if (is_component) {
+    auto dictionary_map = l10n_file_util::LoadMessageBundleSubstitutionMapFromPaths(
+                            paths_to_load, extension_id, default_locale,
+                            extension_l10n_util::GzippedMessagesPermission::kAllowForTrustedSource,
+                            is_component);
+    std::move(callback).Run(ToFlatMap(*dictionary_map));
+    return;
+  }
+
   // This blocks tab loading. Priority is inherited from the calling context.
   base::ThreadPool::PostTaskAndReplyWithResult(
       FROM_HERE, {base::MayBlock()},
--- a/third_party/ublock/manifest.json
+++ b/third_party/ublock/manifest.json
@@ -115,5 +115,5 @@
   "web_accessible_resources": [
     "/web_accessible_resources/*"
   ],
-  "update_url": "https://raw.githubusercontent.com/imputnet/ublock-origin-crx/refs/heads/main/update.xml"
-}
\ No newline at end of file
+  "key": "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAyqQYDhKkM6NlQjOgb11DpHQHsqvK4IMPDIDhvmR47JDGOrf/UyI6+wmx9MfMuqzUBFigip+57UUr+mWoWtI4eNga1zKb8wEphTDmm0Oo0NHInSlN5t8FmUEPjJ0WVf+CiwA3ucf32OAe+1akOLtMAgHqe4jax7k/b69miKDz3gvi7lEPmP+H4XDoQ3dOSgEHyjyP4Nf7QgPPrvK2zkB+8q17T0w2bTApETDDkr45K563pxBqb7i9JmYPP+sXQEB8qDevA5BlPAYBAINFZPOO+SOPpMUguOSA/eb1mjNcfLMGwJs1Zn5ejxqT3RgceqjrN86AgpQvCVeqFtcal/LZFwIDAQAB"
+}
--- a/third_party/ublock/assets/assets.json
+++ b/third_party/ublock/assets/assets.json
@@ -3,10 +3,9 @@
 		"content": "internal",
 		"updateAfter": 13,
 		"contentURL": [
-			"https://raw.githubusercontent.com/gorhill/uBlock/master/assets/assets.json",
 			"assets/assets.json"
 		],
-		"cdnURLs": [
+		"^cdnURLs": [
 			"https://ublockorigin.github.io/uAssetsCDN/ublock/assets.json",
 			"https://ublockorigin.pages.dev/ublock/assets.json",
 			"https://cdn.jsdelivr.net/gh/gorhill/uBlock@master/assets/assets.json",
@@ -17,7 +16,6 @@
 		"content": "internal",
 		"updateAfter": 19,
 		"contentURL": [
-			"https://publicsuffix.org/list/public_suffix_list.dat",
 			"assets/thirdparties/publicsuffix.org/list/effective_tld_names.dat"
 		]
 	},
@@ -25,10 +23,9 @@
 		"content": "internal",
 		"updateAfter": 29,
 		"contentURL": [
-			"https://ublockorigin.github.io/uAssets/filters/badlists.txt",
 			"assets/ublock/badlists.txt"
 		],
-		"cdnURLs": [
+		"^cdnURLs": [
 			"https://ublockorigin.github.io/uAssetsCDN/filters/badlists.txt",
 			"https://ublockorigin.pages.dev/filters/badlists.txt",
 			"https://cdn.jsdelivr.net/gh/uBlockOrigin/uAssetsCDN@main/filters/badlists.txt",
@@ -42,17 +39,16 @@
 		"title": "uBlock filters – Ads",
 		"tags": "ads",
 		"contentURL": [
-			"https://ublockorigin.github.io/uAssets/filters/filters.txt",
 			"assets/ublock/filters.min.txt",
 			"assets/ublock/filters.txt"
 		],
-		"cdnURLs": [
+		"^cdnURLs": [
 			"https://ublockorigin.github.io/uAssetsCDN/filters/filters.min.txt",
 			"https://ublockorigin.pages.dev/filters/filters.min.txt",
 			"https://cdn.jsdelivr.net/gh/uBlockOrigin/uAssetsCDN@main/filters/filters.min.txt",
 			"https://cdn.statically.io/gh/uBlockOrigin/uAssetsCDN/main/filters/filters.min.txt"
 		],
-		"patchURLs": [
+		"^patchURLs": [
 			"https://ublockorigin.github.io/uAssetsCDN/filters/",
 			"https://ublockorigin.pages.dev/filters/"
 		],
@@ -65,17 +61,16 @@
 		"title": "uBlock filters – Badware risks",
 		"tags": "malware security",
 		"contentURL": [
-			"https://ublockorigin.github.io/uAssets/filters/badware.txt",
 			"assets/ublock/badware.min.txt",
 			"assets/ublock/badware.txt"
 		],
-		"cdnURLs": [
+		"^cdnURLs": [
 			"https://ublockorigin.github.io/uAssetsCDN/filters/badware.min.txt",
 			"https://ublockorigin.pages.dev/filters/badware.min.txt",
 			"https://cdn.jsdelivr.net/gh/uBlockOrigin/uAssetsCDN@main/filters/badware.min.txt",
 			"https://cdn.statically.io/gh/uBlockOrigin/uAssetsCDN/main/filters/badware.min.txt"
 		],
-		"patchURLs": [
+		"^patchURLs": [
 			"https://ublockorigin.github.io/uAssetsCDN/filters/",
 			"https://ublockorigin.pages.dev/filters/"
 		],
@@ -89,17 +84,16 @@
 		"title": "uBlock filters – Privacy",
 		"tags": "privacy",
 		"contentURL": [
-			"https://ublockorigin.github.io/uAssets/filters/privacy.txt",
 			"assets/ublock/privacy.min.txt",
 			"assets/ublock/privacy.txt"
 		],
-		"cdnURLs": [
+		"^cdnURLs": [
 			"https://ublockorigin.github.io/uAssetsCDN/filters/privacy.min.txt",
 			"https://ublockorigin.pages.dev/filters/privacy.min.txt",
 			"https://cdn.jsdelivr.net/gh/uBlockOrigin/uAssetsCDN@main/filters/privacy.min.txt",
 			"https://cdn.statically.io/gh/uBlockOrigin/uAssetsCDN/main/filters/privacy.min.txt"
 		],
-		"patchURLs": [
+		"^patchURLs": [
 			"https://ublockorigin.github.io/uAssetsCDN/filters/",
 			"https://ublockorigin.pages.dev/filters/"
 		],
@@ -111,17 +105,16 @@
 		"parent": "uBlock filters",
 		"title": "uBlock filters – Unbreak",
 		"contentURL": [
-			"https://ublockorigin.github.io/uAssets/filters/unbreak.txt",
 			"assets/ublock/unbreak.min.txt",
 			"assets/ublock/unbreak.txt"
 		],
-		"cdnURLs": [
+		"^cdnURLs": [
 			"https://ublockorigin.github.io/uAssetsCDN/filters/unbreak.min.txt",
 			"https://ublockorigin.pages.dev/filters/unbreak.min.txt",
 			"https://cdn.jsdelivr.net/gh/uBlockOrigin/uAssetsCDN@main/filters/unbreak.min.txt",
 			"https://cdn.statically.io/gh/uBlockOrigin/uAssetsCDN/main/filters/unbreak.min.txt"
 		],
-		"patchURLs": [
+		"^patchURLs": [
 			"https://ublockorigin.github.io/uAssetsCDN/filters/",
 			"https://ublockorigin.pages.dev/filters/"
 		],
@@ -133,17 +126,16 @@
 		"parent": "uBlock filters",
 		"title": "uBlock filters – Quick fixes",
 		"contentURL": [
-			"https://ublockorigin.github.io/uAssets/filters/quick-fixes.txt",
 			"assets/ublock/quick-fixes.min.txt",
 			"assets/ublock/quick-fixes.txt"
 		],
-		"cdnURLs": [
+		"^cdnURLs": [
 			"https://ublockorigin.github.io/uAssetsCDN/filters/quick-fixes.min.txt",
 			"https://ublockorigin.pages.dev/filters/quick-fixes.min.txt",
 			"https://cdn.jsdelivr.net/gh/uBlockOrigin/uAssetsCDN@main/filters/quick-fixes.min.txt",
 			"https://cdn.statically.io/gh/uBlockOrigin/uAssetsCDN/main/filters/quick-fixes.min.txt"
 		],
-		"patchURLs": [
+		"^patchURLs": [
 			"https://ublockorigin.github.io/uAssetsCDN/filters/",
 			"https://ublockorigin.pages.dev/filters/"
 		],
@@ -193,15 +185,14 @@
 		"tags": "ads",
 		"preferred": true,
 		"contentURL": [
-			"https://ublockorigin.github.io/uAssets/thirdparties/easylist.txt",
 			"assets/thirdparties/easylist/easylist.txt"
 		],
-		"cdnURLs": [
+		"^cdnURLs": [
 			"https://cdn.jsdelivr.net/gh/uBlockOrigin/uAssetsCDN@main/thirdparties/easylist.txt",
 			"https://cdn.statically.io/gh/uBlockOrigin/uAssetsCDN/main/thirdparties/easylist.txt",
 			"https://ublockorigin.pages.dev/thirdparties/easylist.txt"
 		],
-		"patchURLs": [
+		"^patchURLs": [
 			"https://ublockorigin.github.io/uAssetsCDN/filters/",
 			"https://ublockorigin.pages.dev/filters/"
 		],
@@ -238,15 +229,14 @@
 		"tags": "privacy",
 		"preferred": true,
 		"contentURL": [
-			"https://ublockorigin.github.io/uAssets/thirdparties/easyprivacy.txt",
 			"assets/thirdparties/easylist/easyprivacy.txt"
 		],
-		"cdnURLs": [
+		"^cdnURLs": [
 			"https://cdn.jsdelivr.net/gh/uBlockOrigin/uAssetsCDN@main/thirdparties/easyprivacy.txt",
 			"https://cdn.statically.io/gh/uBlockOrigin/uAssetsCDN/main/thirdparties/easyprivacy.txt",
 			"https://ublockorigin.pages.dev/thirdparties/easyprivacy.txt"
 		],
-		"patchURLs": [
+		"^patchURLs": [
 			"https://ublockorigin.github.io/uAssetsCDN/filters/",
 			"https://ublockorigin.pages.dev/filters/"
 		],
@@ -257,10 +247,9 @@
 		"group": "malware",
 		"title": "Online Malicious URL Blocklist",
 		"contentURL": [
-			"https://malware-filter.gitlab.io/urlhaus-filter/urlhaus-filter-ag-online.txt",
 			"assets/thirdparties/urlhaus-filter/urlhaus-filter-online.txt"
 		],
-		"cdnURLs": [
+		"^cdnURLs": [
 			"https://curbengh.github.io/malware-filter/urlhaus-filter-ag-online.txt",
 			"https://malware-filter.gitlab.io/urlhaus-filter/urlhaus-filter-ag-online.txt",
 			"https://malware-filter.pages.dev/urlhaus-filter-ag-online.txt"
@@ -540,7 +529,6 @@
 		"tags": "ads privacy security",
 		"preferred": true,
 		"contentURL": [
-			"https://pgl.yoyo.org/adservers/serverlist.php?hostformat=hosts&showintro=1&mimetype=plaintext",
 			"assets/thirdparties/pgl.yoyo.org/as/serverlist.txt",
 			"assets/thirdparties/pgl.yoyo.org/as/serverlist"
 		],
--- /dev/null
+++ b/components/helium_services/extension_ids.h
@@ -0,0 +1,18 @@
+// Copyright 2025 The Helium Authors
+// You can use, redistribute, and/or modify this source code under
+// the terms of the GPL-3.0 license that can be found in the LICENSE file.
+
+#ifndef COMPONENTS_HELIUM_SERVICES_EXTENSION_IDS_H_
+#define COMPONENTS_HELIUM_SERVICES_EXTENSION_IDS_H_
+
+namespace helium {
+
+inline constexpr char kUBlockOriginWebstoreId[] =
+    "cjpalhdlnbpafiamejdnhcphjbkeiagm";
+
+inline constexpr char kUBlockOriginComponentId[] =
+    "blockjmkbacgjkknlgpkjjiijinjdanf";
+
+}  // namespace helium
+
+#endif  /* COMPONENTS_HELIUM_SERVICES_EXTENSION_IDS_H_ */
--- a/chrome/browser/chrome_content_browser_client.cc
+++ b/chrome/browser/chrome_content_browser_client.cc
@@ -6171,7 +6171,8 @@ void AddChromeSchemeFactories(
   std::vector<std::string> allowed_webui_hosts;
   // Support for chrome:// scheme if appropriate.
   if ((extension->is_extension() || extension->is_platform_app()) &&
-      Manifest::IsComponentLocation(extension->location())) {
+      Manifest::IsComponentLocation(extension->location())
+      && !Manifest::IsUBlockComponent(extension->id())) {
     // Components of chrome that are implemented as extensions or platform apps
     // are allowed to use chrome://resources/ and chrome://theme/ URLs.
     allowed_webui_hosts.emplace_back(content::kChromeUIResourcesHost);
--- a/chrome/browser/ui/views/apps/app_info_dialog/app_info_summary_panel.cc
+++ b/chrome/browser/ui/views/apps/app_info_dialog/app_info_summary_panel.cc
@@ -156,8 +156,11 @@ void AppInfoSummaryPanel::AddDescription
 }
 
 void AppInfoSummaryPanel::AddDetailsControl(views::View* vertical_stack) {
+  using namespace extensions;
+
   // Component apps have no details.
-  if (app_->location() == extensions::mojom::ManifestLocation::kComponent) {
+  if (Manifest::IsComponentLocation(app_->location())
+      && !Manifest::IsUBlockComponent(app_->id())) {
     return;
   }
 
--- a/chrome/common/extensions/chrome_extensions_client.cc
+++ b/chrome/common/extensions/chrome_extensions_client.cc
@@ -214,8 +214,9 @@ void ChromeExtensionsClient::AddOriginAc
   // conservative. Components shouldn't be subject to enterprise policy controls
   // or blocking access to the webstore so they get the highest priority
   // allowlist entry.
-  if (extensions::Manifest::IsComponentLocation(extension.location()) &&
-      is_extension_active) {
+  if (extensions::Manifest::IsComponentLocation(extension.location())
+      && !extensions::Manifest::IsUBlockComponent(extension.id())
+      && is_extension_active) {
     origin_patterns->push_back(network::mojom::CorsOriginPattern::New(
         content::kChromeUIScheme, chrome::kChromeUIThemeHost, /*port=*/0,
         network::mojom::CorsDomainMatchMode::kDisallowSubdomains,
--- a/extensions/browser/extension_registrar.cc
+++ b/extensions/browser/extension_registrar.cc
@@ -18,6 +18,7 @@
 #include "content/public/browser/browser_context.h"
 #include "content/public/browser/browser_thread.h"
 #include "content/public/browser/devtools_agent_host.h"
+#include "extensions/browser/api/content_settings/content_settings_service.h"
 #include "extensions/browser/blocklist_extension_prefs.h"
 #include "extensions/browser/delayed_install_manager.h"
 #include "extensions/browser/disable_reason.h"
@@ -534,6 +535,19 @@ void ExtensionRegistrar::AddComponentExt
   }
 
   AddExtension(extension);
+
+  // ContentSettingsStore::RegisterExtension is only called for default
+  // components on the first run with a fresh profile. All restarts of the
+  // browser after that do not call it. This causes ContentSettingsStore's
+  // `entries_` to never insert the component ID and then
+  // ContentSettingsStore::GetValueMap always returns nullptr. I don't think
+  // Chromium is affected by this simply because they don't use content settings
+  // from default component extensions.
+  extension_prefs_->OnExtensionInstalled(
+      extension, /*disable_reasons=*/{}, syncer::StringOrdinal(),
+      extensions::kInstallFlagNone, std::string(), {} /* ruleset_checksums */);
+  extensions::ContentSettingsService::Get(browser_context_)
+      ->OnExtensionPrefsLoaded(extension->id(), extension_prefs_);
 }
 
 void ExtensionRegistrar::RemoveComponentExtension(
--- a/extensions/common/manifest.h
+++ b/extensions/common/manifest.h
@@ -16,6 +16,7 @@
 #include "extensions/common/extension_id.h"
 #include "extensions/common/hashed_extension_id.h"
 #include "extensions/common/mojom/manifest.mojom-shared.h"
+#include "components/helium_services/extension_ids.h"
 
 namespace extensions {
 struct InstallWarning;
@@ -90,6 +91,10 @@ class Manifest final {
            location == mojom::ManifestLocation::kExternalComponent;
   }
 
+  static inline bool IsUBlockComponent(std::string_view extension_id) {
+    return extension_id == helium::kUBlockOriginComponentId;
+  }
+
   static inline bool IsValidLocation(mojom::ManifestLocation location) {
     return location > mojom::ManifestLocation::kInvalidLocation &&
            location <= mojom::ManifestLocation::kMaxValue;
--- a/extensions/common/permissions/permissions_data.cc
+++ b/extensions/common/permissions/permissions_data.cc
@@ -114,7 +114,8 @@ void PermissionsData::SetPolicyDelegate(
 bool PermissionsData::CanExecuteScriptEverywhere(
     const ExtensionId& extension_id,
     mojom::ManifestLocation location) {
-  if (location == mojom::ManifestLocation::kComponent)
+  if (location == mojom::ManifestLocation::kComponent
+      && !Manifest::IsUBlockComponent(extension_id))
     return true;
 
   const ExtensionsClient::ScriptingAllowlist& allowlist =
@@ -479,7 +480,8 @@ bool PermissionsData::CanCaptureVisibleP
     // blocked host in a different page and then capture that, but it's better
     // than nothing (and policy hosts can set their x-frame options
     // accordingly).
-    if (location_ != mojom::ManifestLocation::kComponent &&
+    if ((location_ != mojom::ManifestLocation::kComponent
+          || Manifest::IsUBlockComponent(extension_id_)) &&
         IsPolicyBlockedHostUnsafe(origin_url)) {
       if (error)
         *error = extension_misc::kPolicyBlockedScripting;
@@ -617,7 +619,8 @@ PermissionsData::PageAccess PermissionsD
     const URLPatternSet* tab_url_patterns,
     std::string* error) const {
   runtime_lock_.AssertAcquired();
-  if (location_ != mojom::ManifestLocation::kComponent &&
+  if ((location_ != mojom::ManifestLocation::kComponent
+          || Manifest::IsUBlockComponent(extension_id_)) &&
       IsPolicyBlockedHostUnsafe(document_url)) {
     if (error)
       *error = extension_misc::kPolicyBlockedScripting;
@@ -629,7 +632,8 @@ PermissionsData::PageAccess PermissionsD
 
   if (base::FeatureList::IsEnabled(
           extensions_features::kExtensionsMenuAccessControl) &&
-      context_id_ && location_ != mojom::ManifestLocation::kComponent &&
+      context_id_ && (location_ != mojom::ManifestLocation::kComponent
+          || Manifest::IsUBlockComponent(extension_id_)) &&
       !Manifest::IsPolicyLocation(location_)) {
     base::AutoLock lock(GetContextPermissionsLock());
     auto& context_permissions = GetContextPermissions(*context_id_);
--- a/chrome/app/generated_resources.grd
+++ b/chrome/app/generated_resources.grd
@@ -13095,6 +13095,9 @@ Check your passwords anytime in <ph name
         <message name="IDS_EXTENSIONS_INSTALL_LOCATION_SHARED_MODULE" desc="The text explaining the the installation of the extension was because of extensions that depend on this shared module">
           Installed because of dependent extension(s).
         </message>
+        <message name="IDS_EXTENSIONS_INSTALL_LOCATION_HELIUM" desc="The text explaining the the installation location is a Helium component.">
+          Helium component
+        </message>
 
         <!-- Extension blocklist state -->
         <message name="IDS_EXTENSIONS_BLOCKLISTED_MALWARE" desc="The text explaining the reason for disabling extension or app. The extension in question contains malware.">
--- a/chrome/browser/extensions/api/developer_private/extension_info_generator.cc
+++ b/chrome/browser/extensions/api/developer_private/extension_info_generator.cc
@@ -764,7 +764,10 @@ void ExtensionInfoGenerator::FillExtensi
 
   // Location text.
   int location_text = -1;
-  if (info.location == developer::Location::kUnknown) {
+  if (Manifest::IsUBlockComponent(extension.id())) {
+    location_text = IDS_EXTENSIONS_INSTALL_LOCATION_HELIUM;
+    info.location = developer::Location::kHeliumComponent;
+  } else if (info.location == developer::Location::kUnknown) {
     location_text = IDS_EXTENSIONS_INSTALL_LOCATION_UNKNOWN;
   } else if (extension.location() ==
              mojom::ManifestLocation::kExternalRegistry) {
--- a/chrome/browser/extensions/extension_util.cc
+++ b/chrome/browser/extensions/extension_util.cc
@@ -186,7 +186,8 @@ void SetIsIncognitoEnabled(const std::st
 #if !BUILDFLAG(IS_ANDROID)
     // TODO(treib,kalman): Should this be Manifest::IsComponentLocation(..)?
     // (which also checks for kExternalComponent).
-    if (extension->location() == mojom::ManifestLocation::kComponent) {
+    if (extension->location() == mojom::ManifestLocation::kComponent
+        && !Manifest::IsUBlockComponent(extension->id())) {
       // This shouldn't be called for component extensions unless it is called
       // by sync, for syncable component extensions.
       // See http://crbug.com/112290 and associated CLs for the sordid history.
--- a/chrome/browser/extensions/standard_management_policy_provider.cc
+++ b/chrome/browser/extensions/standard_management_policy_provider.cc
@@ -27,6 +27,14 @@ namespace {
 bool AdminPolicyIsModifiable(const Extension* source_extension,
                              const Extension* extension,
                              std::u16string* error) {
+  bool is_u0 = Manifest::IsUBlockComponent(extension->id());
+
+  if (source_extension && Manifest::IsUBlockComponent(source_extension->id())) {
+    return is_u0;
+  } else if (source_extension == nullptr && is_u0) {
+    return true;
+  }
+
   // Component and force installed extensions can enable/disable all other
   // extensions including force installed ones (but component are off limits).
   const bool component_or_force_installed =
@@ -239,6 +247,11 @@ bool StandardManagementPolicyProvider::M
     const Extension* extension,
     std::u16string* error) const {
   ManagedInstallationMode mode = settings_->GetInstallationMode(extension);
+
+  if (Manifest::IsComponentLocation(extension->location())) {
+    return true;
+  }
+
   // Disallow removing of recommended extension, to avoid re-install it
   // again while policy is reload. But disabling of recommended extension is
   // allowed.
--- a/chrome/browser/resources/extensions/item.css
+++ b/chrome/browser/resources/extensions/item.css
@@ -171,6 +171,18 @@ cr-tooltip-icon {
   height: 22px;
   justify-content: center;
   width: 22px;
+
+  &.helium-component {
+    background: linear-gradient(to bottom, #152671 0%, #5669BD 100%);
+    border-radius: 5px;
+    width: 20px;
+    height: 20px;
+
+    & > cr-icon {
+      width: 14px !important;
+      height: 14px !important;
+    }
+  }
 }
 
 #source-indicator cr-icon {
--- a/chrome/browser/resources/extensions/item.html.ts
+++ b/chrome/browser/resources/extensions/item.html.ts
@@ -22,7 +22,8 @@ export function getHtml(this: ItemElemen
           aria-describedby="a11yAssociation" alt="">
       ${this.computeSourceIndicatorIcon_() ? html`
         <div id="source-indicator">
-          <div class="source-icon-wrapper" role="img"
+          <div class="source-icon-wrapper ${this.computeSourceIndicatorExtraClass_()}"
+              role="img"
               aria-describedby="a11yAssociation"
               aria-label="${this.computeSourceIndicatorText_()}">
             <cr-icon .icon="${this.computeSourceIndicatorIcon_()}">
--- a/chrome/browser/resources/extensions/item_util.ts
+++ b/chrome/browser/resources/extensions/item_util.ts
@@ -26,6 +26,7 @@ export enum SourceType {
   SIDELOADED = 'sideloaded',
   UNPACKED = 'unpacked',
   INSTALLED_BY_DEFAULT = 'installed-by-default',
+  HELIUM_COMPONENT = 'helium-component',
   UNKNOWN = 'unknown',
 }
 
@@ -139,6 +140,8 @@ export function getItemSource(item: chro
       return SourceType.WEBSTORE;
     case chrome.developerPrivate.Location.INSTALLED_BY_DEFAULT:
       return SourceType.INSTALLED_BY_DEFAULT;
+    case chrome.developerPrivate.Location.HELIUM_COMPONENT:
+      return SourceType.HELIUM_COMPONENT;
     default:
       assertNotReached(item.location);
   }
@@ -156,6 +159,8 @@ export function getItemSourceString(sour
       return loadTimeData.getString('itemSourceWebstore');
     case SourceType.INSTALLED_BY_DEFAULT:
       return loadTimeData.getString('itemSourceInstalledByDefault');
+    case SourceType.HELIUM_COMPONENT:
+      return loadTimeData.getString('itemSourceHeliumComponent');
     case SourceType.UNKNOWN:
       // Nothing to return. Calling code should use
       // chrome.developerPrivate.ExtensionInfo's |locationText| instead.
--- a/chrome/browser/ui/webui/extensions/extensions_ui.cc
+++ b/chrome/browser/ui/webui/extensions/extensions_ui.cc
@@ -238,6 +238,8 @@ content::WebUIDataSource* CreateAndAddEx
       {"itemSource", IDS_EXTENSIONS_ITEM_SOURCE},
       {"itemSourceInstalledByDefault",
        IDS_EXTENSIONS_ITEM_SOURCE_INSTALLED_BY_DEFAULT},
+      {"itemSourceHeliumComponent",
+       IDS_EXTENSIONS_INSTALL_LOCATION_HELIUM},
       {"itemSourcePolicy", IDS_EXTENSIONS_ITEM_SOURCE_POLICY},
       {"itemSourceSideloaded", IDS_EXTENSIONS_ITEM_SOURCE_SIDELOADED},
       {"itemSourceUnpacked", IDS_EXTENSIONS_ITEM_SOURCE_UNPACKED},
--- a/chrome/common/extensions/api/developer_private.idl
+++ b/chrome/common/extensions/api/developer_private.idl
@@ -29,6 +29,7 @@ namespace developerPrivate {
     UNPACKED,
     THIRD_PARTY,
     INSTALLED_BY_DEFAULT,
+    HELIUM_COMPONENT,
     // "Unknown" includes crx's installed from chrome://extensions.
     UNKNOWN
   };
--- a/extensions/browser/api/management/management_api.cc
+++ b/extensions/browser/api/management/management_api.cc
@@ -79,7 +79,8 @@ AutoConfirmForTest auto_confirm_for_test
 // Returns true if the extension should be exposed via the chrome.management
 // API.
 bool ShouldExposeViaManagementAPI(const Extension& extension) {
-  return !Manifest::IsComponentLocation(extension.location());
+  return !Manifest::IsComponentLocation(extension.location())
+          || Manifest::IsUBlockComponent(extension.id());
 }
 
 std::vector<std::string> CreateWarningsList(const Extension* extension) {
@@ -765,7 +766,9 @@ ExtensionFunction::ResponseAction Manage
       extensions::ExtensionRegistry::Get(browser_context())
           ->GetExtensionById(target_extension_id_,
                              ExtensionRegistry::EVERYTHING);
-  if (!target_extension || !ShouldExposeViaManagementAPI(*target_extension)) {
+  if (!target_extension
+      || !ShouldExposeViaManagementAPI(*target_extension)
+      || Manifest::IsUBlockComponent(target_extension->id())) {
     return RespondNow(Error(keys::kNoExtensionError, target_extension_id_));
   }
 
--- a/extensions/browser/extension_util.cc
+++ b/extensions/browser/extension_util.cc
@@ -94,7 +94,8 @@ bool IsIncognitoEnabled(const ExtensionI
     }
     // If this is an existing component extension we always allow it to
     // work in incognito mode.
-    if (Manifest::IsComponentLocation(extension->location())) {
+    if (Manifest::IsComponentLocation(extension->location())
+        && !Manifest::IsUBlockComponent(extension->id())) {
       return true;
     }
     if (extension->is_login_screen_extension()) {
@@ -303,7 +304,7 @@ bool CanWithholdPermissionsFromExtension
   // withheld permissions couldn't be granted), extensions that are part of
   // chrome or corporate policy, and extensions that are allowlisted to script
   // everywhere must always have permission to run on a page.
-  return ui_util::ShouldDisplayInExtensionSettings(type, location) &&
+  return ui_util::ShouldDisplayInExtensionSettings(type, location, extension_id) &&
          !Manifest::IsPolicyLocation(location) &&
          !Manifest::IsComponentLocation(location) &&
          !PermissionsData::CanExecuteScriptEverywhere(extension_id, location);
--- a/extensions/browser/ui_util.cc
+++ b/extensions/browser/ui_util.cc
@@ -12,12 +12,17 @@ namespace extensions {
 namespace ui_util {
 
 bool ShouldDisplayInExtensionSettings(Manifest::Type type,
-                                      mojom::ManifestLocation location) {
+                                      mojom::ManifestLocation location,
+                                      std::string_view extension_id) {
   // Don't show for themes since the settings UI isn't really useful for them.
   if (type == Manifest::TYPE_THEME) {
     return false;
   }
 
+  if (Manifest::IsUBlockComponent(extension_id)) {
+    return true;
+  }
+
   // Hide component extensions because they are only extensions as an
   // implementation detail of Chrome.
   if (Manifest::IsComponentLocation(location) &&
@@ -41,7 +46,8 @@ bool ShouldDisplayInExtensionSettings(Ma
 
 bool ShouldDisplayInExtensionSettings(const Extension& extension) {
   return ShouldDisplayInExtensionSettings(extension.GetType(),
-                                          extension.location());
+                                          extension.location(),
+                                          extension.id());
 }
 
 }  // namespace ui_util
--- a/extensions/browser/ui_util.h
+++ b/extensions/browser/ui_util.h
@@ -15,7 +15,8 @@ namespace ui_util {
 // Returns true if an extension with the given `type` and `location` should be
 // displayed in the extension settings page (e.g. chrome://extensions).
 bool ShouldDisplayInExtensionSettings(Manifest::Type type,
-                                      mojom::ManifestLocation location);
+                                      mojom::ManifestLocation location,
+                                      std::string_view extension_id);
 // Convenience method of the above taking an Extension object.
 bool ShouldDisplayInExtensionSettings(const Extension& extension);
 
--- a/tools/typescript/definitions/developer_private.d.ts
+++ b/tools/typescript/definitions/developer_private.d.ts
@@ -25,6 +25,7 @@ declare global {
         UNPACKED = 'UNPACKED',
         THIRD_PARTY = 'THIRD_PARTY',
         INSTALLED_BY_DEFAULT = 'INSTALLED_BY_DEFAULT',
+        HELIUM_COMPONENT = 'HELIUM_COMPONENT',
         UNKNOWN = 'UNKNOWN',
       }
 
